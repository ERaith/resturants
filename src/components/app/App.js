import React, {useState,useEffect} from 'react';

import Dashboard from '../pages/Dashboard';

import './App.scss';

function App() {
  const [data,setData] = useState([])

  async function fetchData() {
    const res = await fetch("https://code-challenge.spectrumtoolbox.com/api/restaurants",{
      method:"GET",
      headers:{
        "Content-Type":"application/json",
        "Authorization": "Api-Key q3MNxtfep8Gt",
      }
    })
    return res.json()
    
  }

  useEffect(() => {
    fetchData()
    .then(data=>setData(data))
    
  }, [])

  return (
    <div className="App">
      <Dashboard/>
    </div>
  );
}

export default App;


// fetch(“https://code-challenge.spectrumtoolbox.com/api/restaurants”, {
//   headers: {
//   Authorization: “Api-Key q3MNxtfep8Gt”,
//   },
//  });