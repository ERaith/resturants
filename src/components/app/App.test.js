import { render } from "@testing-library/react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import App from "./App";

test("renders learn react link", () => {
  const wrapper = mount(<App />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
