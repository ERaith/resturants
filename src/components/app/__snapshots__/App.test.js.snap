// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<App>
  <div
    className="App"
  >
    <Dashboard
      data={Array []}
      headerMeta={
        Array [
          Object {
            "key": "name",
            "text": "Resturant Name",
          },
          Object {
            "key": "city",
            "text": "CITY",
          },
          Object {
            "key": "state",
            "text": "STATE",
          },
          Object {
            "key": "telephone",
            "text": "TELEPHONE",
          },
          Object {
            "key": "genre",
            "text": "GENRE",
          },
        ]
      }
    >
      <div
        className="main"
      >
        <div
          className="search-container"
        >
          <SearchBar
            handleSearchChange={[Function]}
          >
            <form
              onSubmit={[Function]}
            >
              <label
                htmlFor="search"
              >
                 SEARCH 
              </label>
              <input
                className="search"
                id="search"
                onChange={[Function]}
                placeholder="Search"
                type="text"
                value=""
              />
              <button
                className="submit-search"
                type="submit"
                value="Submit"
              >
                Submit
              </button>
              <button
                className="clear-search"
                onClick={[Function]}
                type="reset"
              >
                Clear Search
              </button>
            </form>
          </SearchBar>
        </div>
        <div
          className="table-container"
        >
          <Table
            headerMeta={
              Array [
                Object {
                  "key": "name",
                  "text": "Resturant Name",
                },
                Object {
                  "key": "city",
                  "text": "CITY",
                },
                Object {
                  "key": "state",
                  "text": "STATE",
                },
                Object {
                  "key": "telephone",
                  "text": "TELEPHONE",
                },
                Object {
                  "key": "genre",
                  "text": "GENRE",
                },
              ]
            }
            tableData={Array []}
          >
            <table
              className="container"
              id="resturants"
            >
              <tbody>
                <TableHeader
                  headers={
                    Array [
                      Object {
                        "key": "name",
                        "text": "Resturant Name",
                      },
                      Object {
                        "key": "city",
                        "text": "CITY",
                      },
                      Object {
                        "key": "state",
                        "text": "STATE",
                      },
                      Object {
                        "key": "telephone",
                        "text": "TELEPHONE",
                      },
                      Object {
                        "key": "genre",
                        "text": "GENRE",
                      },
                    ]
                  }
                >
                  <tr
                    className="table-row"
                  >
                    <th
                      key="name"
                    >
                      Resturant Name
                    </th>
                    <th
                      key="city"
                    >
                      CITY
                    </th>
                    <th
                      key="state"
                    >
                      STATE
                    </th>
                    <th
                      key="telephone"
                    >
                      TELEPHONE
                    </th>
                    <th
                      key="genre"
                    >
                      GENRE
                    </th>
                  </tr>
                </TableHeader>
                <TableBody
                  data={Array []}
                  headerOrder={
                    Array [
                      "name",
                      "city",
                      "state",
                      "telephone",
                      "genre",
                    ]
                  }
                />
              </tbody>
            </table>
          </Table>
        </div>
      </div>
    </Dashboard>
  </div>
</App>
`;
